modules/sdk-coin-avaxp/src/avaxp.ts

```diff
@@ -52,31 +57,35 @@ export class AvaxP extends BaseCoin {
     return true;
   }
 
-  async verifyTransaction(params: VerifyTransactionOptions): Promise<boolean> {
-    let totalAmount = new BN(0);
-    const coinConfig = coins.get(this.getChain());
-    const { txPrebuild: txPrebuild, txParams: txParams } = params;
-    const rawTx = txPrebuild.txHex;
-    if (!rawTx) {
+  async verifyTransaction(params: AvaxpVerifyTransactionOptions): Promise<boolean> {
+    const txHex = params.txPrebuild && params.txPrebuild.txHex;
+    if (!txHex) {
       throw new Error('missing required tx prebuild property txHex');
     }
-    const transactionBuilder = new AvaxpLib.TransactionBuilderFactory(coinConfig).from(rawTx);
-    const transaction = await transactionBuilder.build();
-    const explainedTx = transaction.explainTransaction();
+    let tx;
+    try {
+      const txBuilder = this.getBuilder().from(txHex);
+      tx = await txBuilder.build();
+    } catch (error) {
+      throw new Error('Invalid transaction');
+    }
+    const explainedTx = tx.explainTransaction();
 
-    if (txParams.recipients !== undefined) {
-      const filteredRecipients = txParams.recipients?.map((recipient) => _.pick(recipient, ['address', 'amount']));
-      const filteredOutputs = explainedTx.outputs.map((output: { address: string; amount: string; memo: string }) =>
-        _.pick(output, ['address', 'amount'])
-      );
+    const { type, stakingOptions, memo } = params.txParams;
+    if (!type || explainedTx.type !== TransactionType[type]) {
+      throw new Error('Tx type does not match with expected txParams type');
+    }
+    if (memo && explainedTx.memo !== memo.value) {
+      throw new Error('Tx memo does not match with expected txParams memo');
+    }
+    if (!params.txParams.recipients || params.txParams.recipients.length === 0) {
```

Comment:
nit: nodeID is not really a recipient. 
Can we verify based on the tx type? We should only check `stakingOptions` when the tx is for a staking operation